Code/server/GameServer.cpp

```diff
@@ -383,6 +391,129 @@ void GameServer::BindServerCommands()
+                        {
+                            AddAdminSession(apPlayer->GetConnectionId());
+                            out->info("{} admin privileges added", backupUsername.c_str());
+                            playerFound = true;
+                            return;
+                        }
+                    });
+
+                }
+                if (!playerFound)
+                    out->warn("{} is not an admin", backupUsername.c_str());
+            }
+        });
+    m_commands.RegisterCommand<std::string>(
+        "RemoveAdmin", "Remove admin privileges from player", [&](Console::ArgStack& aStack) {
+            auto out = spdlog::get("ConOut");
+
+            const auto& cUsername = aStack.Pop<String>();
+            bool playerFound = false;
+
+            for (const auto& cAdmin : m_adminSessions)
+            {
+                Player* pPlayer = PlayerManager::Get()->GetByConnectionId(cAdmin);
+                if (pPlayer->GetUsername() == cUsername)
+                {
+                    RemoveAdminSession(pPlayer->GetConnectionId());
+                    out->info("{} admin privileges revoked", cUsername.c_str());
+                    playerFound = true;
+                    break;
+                }
+            }
+
+            if (!playerFound)
+            {
+                // space in username handling
+                String backupUsername = cUsername;
+                if (cUsername.find('_') != std::string::npos)
+                {
+
+                    while (backupUsername.find('_') != std::string::npos)
+                    {
+                        std::ranges::replace(backupUsername, '_', ' ');
+                    }
+
+                    for (const auto& cAdmin : m_adminSessions)
+                    {
+                        Player* pPlayer = PlayerManager::Get()->GetByConnectionId(cAdmin);
+                        if (pPlayer->GetUsername() == backupUsername)
+                        {
+                            RemoveAdminSession(pPlayer->GetConnectionId());
+                            out->info("{} admin privileges revoked", backupUsername.c_str());
+                            playerFound = true;
+                            break;
+                        }
+                    }
+
+                }
+                if (!playerFound)
+                    out->warn("{} is not an admin", backupUsername.c_str());
+            }
+        });
+    m_commands.RegisterCommand<>(
+        "admins", "List all admins", [&](Console::ArgStack&) {
+            auto out = spdlog::get("ConOut");
+            if (m_adminSessions.size() == 0)
+            {
+                out->warn("No admins");
+                return;
+            }
+
+            String output = "Admins: ";
+            bool _first = true;
+
+            for (const auto& cAdmin : m_adminSessions)
+            {
+                const auto& cUsername = PlayerManager::Get()->GetByConnectionId(cAdmin)->GetUsername();
+
```

Comment:
Even though unlikely, we should still handle if GetConnectionId somehow returns null

Reply:
Now has a nullptr check and prints to console if m_adminSessions contains a session it shouldn't